docs/tips-and-tricks/debugging-insights.md

```diff
@@ -0,0 +1,175 @@
+---
+sidebar_position: 1
+slug: /community-wisdom/debugging-insights
+sidebar_label: 'Debugging Insights'
+doc_type: 'how-to-guide'
+keywords: [
+  'clickhouse troubleshooting',
+  'clickhouse errors',
+  'slow queries',
+  'memory problems', 
+  'connection issues',
+  'performance optimization',
+  'database errors',
+  'configuration problems',
+  'debug',
+  'solutions'
+]
+title: 'Lessons - debugging insights'
+description: 'Find solutions to the most common ClickHouse problems including slow queries, memory errors, connection issues, and configuration problems.'
+---
+
+# ClickHouse operations: community debugging insights {#clickhouse-operations-community-debugging-insights}
+*This guide is part of a collection of findings gained from community meetups. For more real world solutions and insights you can [browse by specific problem](./community-wisdom.md).*
+*Suffering from high operational costs? Check out the [Cost Optimization](./cost-optimization.md) community insights guide.*
+
+## Essential system tables {#essential-system-tables}
+
+These system tables are fundamental for production debugging:
+
+### system.errors {#system-errors}
+
+Shows all active errors in your ClickHouse instance.
+
+```sql
+SELECT name, value, changed 
+FROM system.errors 
+WHERE value > 0 
+ORDER BY value DESC;
+```
+
+### system.replicas {#system-replicas}
+
+Contains replication lag and status information for monitoring cluster health.
+
+```sql
+SELECT database, table, replica_name, absolute_delay, queue_size, inserts_in_queue
+FROM system.replicas 
+WHERE absolute_delay > 60
+ORDER BY absolute_delay DESC;
+```
+
+### system.replication_queue {#system-replication-queue}
+
+Provides detailed information for diagnosing replication problems.
+
+```sql
+SELECT database, table, replica_name, position, type, create_time, last_exception
+FROM system.replication_queue 
+WHERE last_exception != ''
+ORDER BY create_time DESC;
+```
+
+### system.merges {#system-merges}
+
+Shows current merge operations and can identify stuck processes.
+
+```sql
+SELECT database, table, elapsed, progress, is_mutation, total_size_bytes_compressed
+FROM system.merges 
+ORDER BY elapsed DESC;
+```
+
+### system.parts {#system-parts}
+
+Essential for monitoring part counts and identifying fragmentation issues.
+
+```sql
+SELECT database, table, count() as part_count
+FROM system.parts 
+WHERE active = 1
+GROUP BY database, table
+ORDER BY count() DESC;
+```
+
+## Common production issues {#common-production-issues}
+
+### Disk space problems {#disk-space-problems}
+
+Disk space exhaustion in replicated setups creates cascading problems. When one node runs out of space, other nodes continue trying to sync with it, causing network traffic spikes and confusing symptoms. One community member spent 4 hours debugging what was simply low disk space.
```

Comment:
Do we have a simple query we can share here to detect remaining disk space on each node?